### ISSUE TITLE
Trouble installing through VScode on macOS

### ISSUE THREAD

**Issue by johnwickerson** [dafny-lang/dafny#2869]
Hi, when I open a `.dfy` file in VS Code, I get the following error as it tries to install Dafny...

> Starting Dafny installation
> deleting previous Dafny installation at /Users/jpw48/.vscode/extensions/dafny-lang.ide-vscode-2.8.3/out/resources/3.8.1
> Found a non-supported architecture OSX:arm64. Going to install from source.
> Installing Dafny from source in /Users/jpw48/.vscode/extensions/dafny-lang.ide-vscode-2.8.3/out/resources/3.8.1/custom/arm64.
> 
> Executing: brew install dotnet-sdk
> Executing: javac -version
> Javac not found. Please install a valid JDK and ensure that the path containing javac is in the PATH environment variable. You can obtain a free open-source JDK 1.8 from here: https://aws.amazon.com/corretto/

The thing is, if I open the Terminal tab in VS Code, and execute `javac -version` manually, I get 
> javac 19

So I'm not sure why it's complaining about not being able to find a JDK. Any thoughts? Thanks very much.

By the way, I'm running macOS 12.6 (Monterey) on an Apple M1 Pro.

**Comment by johnwickerson** [comment]
Quick update on this: it's happier if I install version 1.8 of the JDK (which is, to be fair, what the error message suggests). But it then gets stuck trying to find the right version of `dotnet` -- it seems to want version 6.0.0 exactly. Am I doing something wrong? It seems surprisingly complicated to get Dafny running on a vanilla Mac. Is anybody else struggling?

**Comment by kjx** [comment]
I had somewhat similar problems which are logged somewhere, I think on the IDE git.
If you use homebrew
-  `brew install dafny`  should get a stable dotnet and dafny.
- do a standard install of VSCODE.
- start VSCode, install the dafny plugin, 
- configure the VSCode plugin to use the homebrew dafny. I have
    - Compiler Runtime Path: /opt/homebrew/Cellar/dafny/3.9.0/libexec/Dafny.dll
    - Language Server Runtime Path: /opt/homebrew/Cellar/dafny/3.9.0/libexec/DafnyLanguageServer.dll

**Comment by prvshah51** [comment]
Can you try choosing v3.9 from extension setting and let it install it for you ? and if v3.8.1 is latest one it shows there is an update for the extension you can install.

**Comment by prvshah51** [comment]
Also, just released v2.8.4 for VSCode extension which will install Dafny v3.9.0 by default.

**Comment by kjx** [comment]
so I tried clearing paths to daftly, reinstalling 284, same problem as I had earlier #dafny-lang/ide-vscode/283

the presenting problems may be different but the underlying issue here is not getting arm64 (for either Java or dafny proper) on M* macs...

**Comment by johnwickerson** [comment]
Thank you very much @kjx and @prvshah51 for your help. (Sorry for the slow reply; I've been on leave.)

I've now got it working, probably thanks to the updates that have been done by the Dafny team in the meantime. What worked for me was something roughly like this:
1. Running `brew install dotnet` (which installed version 6.0.110 of the .NET SDK)
2. Appending `export DOTNET_ROOT="/opt/homebrew/opt/dotnet/libexec"` to my `~/.zshrc` (as instructed by homebrew)
3. Uninstalling and reinstalling the Dafny extension in VS Code.

(I ran various other commands like `brew install dafny` but I'm not sure which ones actually helped!)
